python/whylogs/core/resolvers.py

```diff
@@ -16,11 +16,85 @@
 ColumnSchema: TypeAlias = "ColumnSchema"  # type: ignore
 
 
+@dataclass
+class MetricSpec:
+    """
+    Specify a Metric to instantiate.
+    """
+
+    metric: Any  # Should be a subclass of Metric, it should be the class, not an instance
+    config: Optional[MetricConfig] = None  # omit to use default MetricConfig
+
+
+@dataclass
+class ResolverSpec:
+    """
+    Specify the metrics to instantiate for matching columns. column_name
+    takes precedence over column_type. column_type should be a subclass
+    of DataType, i.e., AnyType, Frational, Integral, or String. Pass the
+    class, not an instance.
+    """
+
+    column_name: Optional[str] = None  # TODO: maybe make this a regex
+    column_type: Optional[Any] = None
+    metrics: List[MetricSpec] = field(default_factory=list)
+
+
 class Resolver(ABC):
     """A resolver maps from a column name and a data type to trackers.
 
     Note that the key of the result dictionaries defines the namespaces of the metrics in the serialized form."""
 
+    def __init__(self):
+        self.additional_specs: List[ResolverSpec] = []
+
+    def add_resolver_spec(self, resolver_spec: ResolverSpec):
+        """Add resolver specification to existing resolver instance.
+
+        Parameters
+        ----------
+        resolver_spec : ResolverSpec
+            Resolver specification that define the metrics to instantiate for matching columns.
+        """
+        if resolver_spec.column_name and resolver_spec.column_type:
+            logger.warning(
```

Comment:
We should move the `ResolverSpec` validation into `ResolverSpec::__post_init__()`.

Reply:
Now that I'm using `DeclarativeResolver` to resolve the additional metrics, I removed this validation. I agree that the validation from `DeclarativeResolver` could be moved to `ResolverSpec::__post_init__()`.

I'm ok with doing this in a follow-up, though. lmk if that's ok

Reply:
Make a click ticket to track it :)